src/NuGet.Clients/NuGet.CommandLine/Common/MSBuildProjectSystem.cs

```diff
@@ -46,17 +48,71 @@ public NuGetFramework TargetFramework
         {
             get
             {
-                string moniker = GetPropertyValue("TargetFrameworkMoniker");
+                string moniker = GetTargetFrameworkString();
+
                 if (String.IsNullOrEmpty(moniker))
                 {
                     return null;
                 }
-                return NuGetFramework.Parse(moniker);
+
+                var framework = NuGetFramework.Parse(moniker);
+
+                // if the framework is .net core 4.5.1 return windows 8.1
+                if (framework.Framework.Equals(FrameworkConstants.FrameworkIdentifiers.NetCore)
+                    && framework.Version.Equals(Version.Parse("4.5.1.0")))
+                {
+                    return new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Windows, Version.Parse("8.1"), framework.Profile);
```

Comment:
Makes sense, please work with Justin & Joel on the design

From: Ashish Jain [mailto:notifications@github.com]
Sent: Thursday, June 9, 2016 9:31 AM
To: NuGet/NuGet.Client <NuGet.Client@noreply.github.com>
Cc: Yishai Galatzer <yigalatz@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [NuGet/NuGet.Client] Setting right TFM when running nuget.exe update for native or js projects (#662)


In src/NuGet.Clients/NuGet.CommandLine/Common/MSBuildProjectSystem.cs<https://github.com/NuGet/NuGet.Client/pull/662#discussion_r66475134>:

>                  if (String.IsNullOrEmpty(moniker))

>                  {

>                      return null;

>                  }

> -                return NuGetFramework.Parse(moniker);

> +

> +                var framework = NuGetFramework.Parse(moniker);

> +

> +                // if the framework is .net core 4.5.1 return windows 8.1

> +                if (framework.Framework.Equals(FrameworkConstants.FrameworkIdentifiers.NetCore)

> +                    && framework.Version.Equals(Version.Parse("4.5.1.0")))

> +                {

> +                    return new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Windows, Version.Parse("8.1"), framework.Profile);

currently it does not exist any where in Frameworks, but i can add a new api something like GetSupportedFramework(framwork) in NugetFramework itself so that the same api could be used across vs, commandline,

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/NuGet/NuGet.Client/pull/662/files/30a8b8345f6963f19927ed7c90ca4112670cc357#r66475134>, or mute the thread<https://github.com/notifications/unsubscribe/ABLmt9mWDwF6uSjiR8U__hDeQuMJnJKqks5qKD-_gaJpZM4IxhO6>.